### DISCUSSION TITLE
x509: certificate signed by unknown authority when running okteto build

### DISCUSSION THREAD

**Discussion by bangbaew** [okteto/okteto#3744] [Q&A]
![image](https://github.com/okteto/okteto/assets/29559559/89b424a3-75dd-4434-966e-0c5c32133115)

when I run `okteto deploy` or `okteto build` from my local machine after logging in to okteto self hosted context, it always ends up with `"transport: authentication handshake failed: x509: certificate signed by unknown authority"`, i use Cloudflare as a domain and used Cloudflare api key for the certificate, i've done all my installation tasks, but I if click deploy and choose a repo from GitHub, the build process works fine.
![image](https://github.com/okteto/okteto/assets/29559559/a35ad3e8-3af3-4c45-9579-6d31f6ee2ff3)

Can someone help me with this?

**Comment by nitishfy** [comment]
Hi @bangbaew , thanks for putting this up!

I think the error message you're encountering, _transport: authentication handshake failed: x509: certificate signed by unknown authority,_ indicates that the certificate used by your self-hosted Okteto environment is not recognized as a trusted authority on your local machine.

Try adding the custom certificate to your local machine's trust store and validate if it works fine.

Let me know if that works for you.

**Comment by pchico83** [comment]
@bangbaew could you post your question in our https://community.okteto.com/ forum? this repo is only for Okteto CLI issues
